File: ChimpokomonPython/ataque/GarraMecanica.py
from Ataque import Ataque
from chimpokomon.Chimpokomon import Chimpokomon
import random
import Ataque.Ataque;
import chimpokomon.Chimpokomon;

class GarraMecanica(Ataque):
    def dañar(self,  chinpokomon : Chimpokomon, chimpokomonAtacante : Chimpokomon):
        ram = random.randint(0,1)
        if ram < 0.5:
            chinpokomon.restarVida(chimpokomon.getNivelDeVida() * 0.5)
            print(chimpokomonAtacante.getNombre() + " Realizo un ataque critico")
        
        else:
            chinpokomon.restarVida(2)

File: ChimpokomonPython/Ataque/GarraMecanica.java
package Ataque;

import Ataque.Ataque;
import chimpokomon.Chimpokomon;

public class GarraMecanica extends Ataque {
    @Override
    public void dañar(Chimpokomon chinpokomon, Chimpokomon chimpokomonAtacante) {
        double random = Math.random();
        if(random < 0.1) {
            chinpokomon.restarVida((int) (chinpokomon.getNivelDeVida() * 0.5));
            System.out.println(chinpokomon.getNombre() + " Realizo un ataque critico");
        }
        else {
            chinpokomon.restarVida(2);
        }

    }
}
